Fix #17074: Removed the expand/collapse button when interaction isn't added

AdityaRaimec22 opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #17074 
2. This PR does the following: Removes the expand/collapse button when interaction isn't added

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## Proof that changes are correct
<!--
Add videos/screenshots of the user-facing interface to demonstrate that the changes
made in this PR work correctly. If this PR is for a developer-facing feature,
provide the videos/screenshots of developer-facing interface.

Please also include videos/screenshots of the developer tools
browser console, so that we can be sure that there are no console errors.

If the changes in your PRs are autogenerated via a script and you cannot provide proof
for the changes then please leave a comment "No proof of changes needed because {{Reason}}"
and remove all the sections below.
-->

#### Proof of changes on desktop with slow/throttled network

![englishDesk](https://user-images.githubusercontent.com/118007662/220025101-8bb7d94e-f815-44dd-95cb-b6cce5eaa41e.png)

<!--
Make sure to properly verify that everything works correctly, and that there are
no weird UI mistakes or other problems. Also, if there are any newly added fields,
try to fill them out and test that different inputs are correctly accepted/rejected.

Throttle the network (to 3G) using the browser Developer Tools (see references below).
There should be no performance or UI issues while the network is slow.

References:
 - Chrome: https://css-tricks.com/throttling-the-network/
 - Firefox: https://developer.mozilla.org/en-US/docs/Tools/Network_Monitor/Throttling
-->

#### Proof of changes on mobile phone

![mobile](https://user-images.githubusercontent.com/118007662/220025231-5edf0602-0382-4b10-a18b-71dd72111cd8.png)


<!--
In some cases this is not needed (e.g. for pages that we do not expect to
support mobile phones, or for backend-only features).

Feel free to use the Developer Tools emulator for this.

References:
 - Chrome: https://developer.chrome.com/docs/devtools/device-mode/
 - Firefox: https://firefox-source-docs.mozilla.org/devtools-user/index.html#responsive-design-mode
-->

#### Proof of changes in Arabic language

![arabiDesk](https://user-images.githubusercontent.com/118007662/220025264-362b93c4-a231-4bf4-8324-ab5a4a7a02dd.png)


<!--
If the PR changes the UI, make sure to add screenshots with the site
language set to Arabic as well (we use Arabic as it is a language written from right to left).
-->

## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Make-a-pull-request).
- If you need a review or an answer to a question, and don't have permissions to assign people, **leave a comment** like the following: "{{Question/comment}} @{{reviewer_username}} PTAL". Oppiabot will help assign that person for you.
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia's-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.
- Some of the e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-CI-checks-fail-on-your-PR).

AdityaRaimec22 committed:
removed the WHO link from OPPIA Avatar

AdityaRaimec22 committed:
Merge branch 'develop' of https://github.com/AdityaRaimec22/oppia into develop

AdityaRaimec22 committed:
removed WHO link

AdityaRaimec22 committed:
first commit

AdityaRaimec22 committed:
First Commit

AdityaRaimec22 committed:
another commit

AdityaRaimec22 committed:
frist commit

AdityaRaimec22 committed:
corrected changes

AdityaRaimec22 committed:
another

AdityaRaimec22 commented:
@Lawful2002 can you please rerun the failed tests because they failed for no reason.

Lawful2002 commented on core/templates/components/state-editor/state-interaction-editor/state-interaction-editor.component.html:
```diff
@@ -6,14 +6,6 @@
       <h3 class="oppia-exp-interaction-card-header">
         Interaction <span *ngIf="getCurrentInteractionName()">( {{getCurrentInteractionName()}} )</span>
       </h3>
-      <i class="fa fa-caret-down"
-         *ngIf="!interactionEditorIsShown"
-         aria-hidden="true">
-      </i>
-      <i class="fa fa-caret-up"
-         *ngIf="interactionEditorIsShown"
-         aria-hidden="true">
-      </i>
```
What happens when an interaction is present? I think you have completely removed the dropfown icon, instead of hiding it only when there are no interactions.

AdityaRaimec22 replied:
I think the job is done when Interaction is added the expand/collapse icon will reappear

Lawful2002 reviewed (changes requested):
Thanks @AdityaRaimec22, I have left a comment. PTAL. Also, please upload a video showing this functionality both when an interaction is present and when it isn't.

AdityaRaimec22 commented:
@Lawful2002 can you please show me how interaction can be made available at the interaction box.

![image](https://user-images.githubusercontent.com/118007662/220260678-7b8f3973-0149-485d-b92f-86c5337adbde.png)

Because whenever I add an interaction , after successfully adding it I automatically exits and again reaches to the submit question tab. And when I click the same suggest question there opens a tab for inputting a new question.

That's why only I thought there's no need of that toggle button because I was able to add only one interaction and in that also a new section opens up.

seanlip commented:
@AdityaRaimec22 This "auto-exiting" behaviour you describe is weird. Could you please attach a video that demonstrates it? Are you saying that after clicking the "Add Interaction" button, the modal dialog closes and brings you back to the main questions submission page? If so, that should not be happening, but it would help to see exactly what you're describing so that we're all on the same page.

AdityaRaimec22 commented:
> @AdityaRaimec22 This "auto-exiting" behaviour you describe is weird. Could you please attach a video that demonstrates it? Are you saying that after clicking the "Add Interaction" button, the modal dialog closes and brings you back to the main questions submission page? If so, that should not be happening, but it would help to see exactly what you're describing so that we're all on the same page.

@seanlip @Lawful2002  What I mean is depicted below

https://user-images.githubusercontent.com/118007662/220352276-e6e48098-0b5d-44d7-b6ff-6f9c37263361.mp4

Above shown is I added the Interaction. But after adding it when I again click the suggest question there opens the same window (It again asks for input a new question and new interaction) and the old one that I added requires a review but I am unable to review it as described by me in the discussion section.


https://user-images.githubusercontent.com/118007662/220353252-28225681-9dc4-435e-bf77-d89f08dd7830.mp4

So, the conclusion is that I am unable to show that after adding interaction the expand/collapse button reappears.
Hence I completely deleted that button in my PR because I thought it is of no use as when we click Add interaction there opens a new tab and the expand/collapse button remains of no use.

If I am wrong can you please show me how can I do the testing of the issue #17074 . So, that I can make new changes and commit them here and then only I'll be able to show the proof.

seanlip commented:
OK, two things:

(a) After you submit a question, it goes into the review queue, and clicking "Suggest question" will result in a brand-new submission. As mentioned in the discussion thread, you cannot review your own submissions.

(b) To test the interaction thing, when creating a question and after selecting an interaction, close the "add response" modal when it comes up. Then you should have the opportunity to delete the interaction when scrolling up in the dialog box. This is at 0:07 of your first video.

AdityaRaimec22 committed:
Merge branch 'develop' into Branch1

AdityaRaimec22 committed:
another commit

AdityaRaimec22 committed:
one more commit

AdityaRaimec22 committed:
Last commit
